swc_common and serde Dependency Error when installing red on windows

Error:

   Compiling swc_common v9.2.0
error[E0432]: unresolved import `serde::__private`
 --> C:\Users\arifa\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_common-9.2.0\src\private\mod.rs:3:9
  |
3 | pub use serde::__private as serde;
  |         ^^^^^^^---------^^^^^^^^^
  |         |      |
  |         |      help: a similar name exists in the module: `private`
  |         no `__private` in the root

   Compiling minimal-lexical v0.2.1
   Compiling crc32fast v1.5.0
   Compiling nom v7.1.3
   Compiling par-iter v1.0.2
For more information about this error, try `rustc --explain E0432`.
error: could not compile `swc_common` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `red v0.1.0 (C:\Users\arifa\red)`, intermediate artifacts can be found at `C:\Users\arifa\red\target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

--------------------------------------------------------------------------------------------------------------------------------------------------

I ran "rustc --explain E0432" and the results is:
red on  master is 󰏗 v0.1.0 via 󱘗 v1.91.0 took 18m24s
✦ ❯ rustc --explain E0432
An import was unresolved.

Erroneous code example:

```
use something::Foo; // error: unresolved import `something::Foo`.
```

In Rust 2015, paths in `use` statements are relative to the crate root. To
import items relative to the current and parent modules, use the `self::` and
`super::` prefixes, respectively.

In Rust 2018 or later, paths in `use` statements are relative to the current
module unless they begin with the name of a crate or a literal `crate::`, in
which case they start from the crate root. As in Rust 2015 code, the `self::`
and `super::` prefixes refer to the current and parent modules respectively.

Also verify that you didn't misspell the import name and that the import exists
in the module from where you tried to import it. Example:

```
use self::something::Foo; // Ok.

mod something {
    pub struct Foo;
}
```

If you tried to use a module from an external crate and are using Rust 2015,
you may have missed the `extern crate` declaration (which is usually placed in
the crate root):

```
extern crate core; // Required to use the `core` crate in Rust 2015.

use core::any;
```

Since Rust 2018 the `extern crate` declaration is not required and
you can instead just `use` it:

```
use core::any; // No extern crate required in Rust 2018.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@ async-trait = "0.1.77"
 bon = "3.3.2"
 clap = { version = "4.5.26", features = ["derive"] }
 crossterm = { version = "0.27.0", features = ["event-stream"] }
-deno_ast = { version = "0.48.0", features = ["transpiling"] }
+deno_ast = { version = "=0.53.2", features = ["transpiling"] }
 deno_core = "0.320.0"
 futures = "0.3.30"
 futures-timer = "3.0.2"

--- a/src/plugin/runtime.rs
+++ b/src/plugin/runtime.rs
@@ -805,6 +805,8 @@ extension!(
 
 #[cfg(test)]
 mod tests {
+    use std::fs;
+
     use crate::editor::Action;
 
     use super::*;
@@ -838,6 +840,51 @@ mod tests {
             )
             .await
             .unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_runtime_file_typescript_import_transpiles() -> anyhow::Result<()> {
+        let temp_dir =
+            std::env::temp_dir().join(format!("red-typescript-plugin-{}", uuid::Uuid::new_v4()));
+        fs::create_dir_all(&temp_dir)?;
+
+        let module_path = temp_dir.join("typed-plugin.ts");
+        fs::write(
+            &module_path,
+            r#"
+                export type PluginInfo = {
+                    name: string;
+                    commandCount: number;
+                };
+
+                const plugin: PluginInfo = {
+                    name: "red",
+                    commandCount: 41,
+                };
+
+                export const commandCount: number = plugin.commandCount + 1;
+                export default plugin.name;
+            "#,
+        )?;
+
+        let module_specifier = deno_core::ModuleSpecifier::from_file_path(&module_path)
+            .map_err(|_| anyhow::anyhow!("failed to create module specifier"))?;
+
+        let mut runtime = Runtime::new();
+        let result = runtime
+            .add_module(&format!(
+                r#"
+                    import pluginName, {{ commandCount }} from "{module_specifier}";
+
+                    if (pluginName !== "red" || commandCount !== 42) {{
+                        throw new Error(`unexpected plugin export: ${{pluginName}}/${{commandCount}}`);
+                    }}
+                "#
+            ))
+            .await;
+
+        fs::remove_dir_all(&temp_dir)?;
+        result
     }
 
     #[tokio::test]

